### PR TITLE
Improvements to resource loading and dependencies.

### PR DESCRIPTION
**Pull request by careo**
* Added Resource.property
  * Added IdentityMap#add_selectors
  * Support :* and '*' for selects in queries

Signed-off-by: Dane Jensen <dane@rightscale.com>

### CODE REVIEW

File: lib/praxis-mapper/selector_generator.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    def add_association(resource, name, field)
+      association = resource.model.associations.fetch(name) do
+        raise "missing association for #{resource} with name #{name}"
+      end
+      associated_resource = resource.model_map[association[:model]]
+
+      add_track(resource, name)
+
+      # TODO: flesh out possible association types we should handle here
```
Comment (blanquer) [inline lib/praxis-mapper/selector_generator.rb]: We need to address this comment. At the very least to think about the implications.

Comment (careo) [inline lib/praxis-mapper/selector_generator.rb]: Yep. many_to_many is in, which is the last of the native sequel associations we support.

File: lib/praxis-mapper/query/base.rb [hunk 2]
```diff
@@ -51,11 +67,18 @@ def connection
       # @example select(:account_id, "user_id", {"create_time" => :created_at})
       def select(*fields)
         if fields.any?
-          @select = {} if @select.nil?
+          return @select if @select == true
+
+          @select = {} if @select.nil? # FIXME: prepopulate this with identities
```
Comment (careo) [inline lib/praxis-mapper/query/base.rb]: Should probably get this in. We didn't do it before, but it's more likely to become an issue at some point.

**Comment by blanquer** [comment]
LGTM

**Pull request merged**